packages/builder/src/stores/portal/environment.ts

```diff
@@ -0,0 +1,79 @@
+import { get } from "svelte/store"
+import { API } from "@/api"
+import { licensing } from "@/stores/portal"
+import { BudiStore } from "../BudiStore"
+import {
+  CreateEnvironmentVariableRequest,
+  EventPublishType,
+  StatusEnvironmentVariableResponse,
+  UpdateEnvironmentVariableRequest,
+} from "@budibase/types"
+
+type EnvVar = {
+  name: string
+}
+
+interface EnvironmentState {
+  variables: EnvVar[]
+  status: StatusEnvironmentVariableResponse
+}
+
+class EnvironmentStore extends BudiStore<EnvironmentState> {
+  constructor() {
+    super({
+      variables: [],
+      status: {
+        encryptionKeyAvailable: false,
+      },
+    })
+  }
+
+  async checkStatus() {
+    const status = await API.checkEnvironmentVariableStatus()
+    this.update(store => {
+      store.status = status
+      return store
+    })
+  }
+
+  async loadVariables() {
+    if (get(licensing).environmentVariablesEnabled) {
+      const envVars: string[] = (await API.fetchEnvironmentVariables())
```

Comment:
NAB: Don't think we need to type this do we? `string[]` will be inferred from the API? I could be wrong

Reply:
We do actually, only because the API response type is a custom thing (something along the lines of FetchEnvironvmentVariablesResponse) which is itself just aliased to a `string[]`, but I'd rather store a normal string array in the store instead of the response type, hence the casting here.